client/hotrod-client/src/test/java/org/infinispan/client/hotrod/impl/iteration/BaseMultiServerRemoteIteratorTest.java

```diff
@@ -214,17 +214,17 @@ public void testRetrieveMetadata() {
       }
    }
 
-   static final class ToHexConverterFactory implements KeyValueFilterConverterFactory<Object, Integer, String>, Serializable {
+   static final class ToHexConverterFactory implements KeyValueFilterConverterFactory<Object, Object, String>, Serializable {
       @Override
-      public KeyValueFilterConverter<Object, Integer, String> getFilterConverter() {
+      public KeyValueFilterConverter<Object, Object, String> getFilterConverter() {
          return new HexFilterConverter();
       }
 
       @ProtoName("HexFilterConverter")
-      static class HexFilterConverter extends AbstractKeyValueFilterConverter<Object, Integer, String> {
+      static class HexFilterConverter extends AbstractKeyValueFilterConverter<Object, Object, String> {
          @Override
-         public String filterAndConvert(Object key, Integer value, Metadata metadata) {
-            return Integer.toHexString(value);
+         public String filterAndConvert(Object key, Object value, Metadata metadata) {
+            return Integer.toHexString(Integer.parseInt(value.toString()));
```

Comment:
Have to admit this code does not sit well for me, but I guess if it is needed solely for tests with text/plain that is okay.